modules/rte_mis_school/rte_mis_school.module

```diff
@@ -615,42 +614,87 @@ function rte_mis_school_form_level_education_details_validate(array &$form, Form
       }
     }
   }
+}
 
-  // Iterate over the existing entity values (existing paragraphs).
-  if (!empty($entity_details)) {
-    foreach ($entity_details as $educational_detail_entity) {
-      // Extract the values from the entity.
-      $board_type = $educational_detail_entity->get('field_board_type')->value;
-      $category = $educational_detail_entity->get('field_education_type')->value;
-      $medium = $educational_detail_entity->get('field_medium')->value;
-      $education_level = $educational_detail_entity->get('field_education_level')->value;
-
-      // Check for conflicting education levels.
-      $combination = $board_type . '|' . $category . '|' . $medium;
-      if (isset($education_levels_checked[$combination]) && $education_levels_checked[$combination] == $education_level) {
-        if (in_array($education_level, $education_levels_checked[$combination])) {
-          $form_state->setErrorByName("field_education_details][$key][subform][field_education_level", t('Different entries of Education Level are not allowed for the same Board Type, Education Type, and Medium.'));
-          return;
-        }
-      }
-      $education_levels_checked[$combination][] = $education_level;
+/**
+ * Helper function to check for conflicts in education levels.
+ *
+ * @param array $education_levels_checked
+ *   An array that tracks the education levels for each category.
+ * @param string $board_type
+ *   The board type to check.
+ * @param string $category
+ *   The education type category.
+ * @param string $medium
+ *   The medium of school.
+ *
+ * @return string
+ *   Returns Status of the check.
+ */
+function rte_mis_school_check_education_details(array &$education_levels_checked, $board_type, $category, $medium) {
+
+  // Create a unique key for the current entry.
+  $key_for_conflict_check = $board_type . '|' . $category . '|' . $medium;
+
+  // Check for duplicate entries of the
+  // combination (board_type|category|medium).
+  if (in_array($key_for_conflict_check, $education_levels_checked)) {
+    return 'Duplicate';
+  }
+
+  // Handle 'co-ed' conflicts.
+  if ($category === 'co-ed') {
+    // Check if 'boys' or 'girls' already exist for the
+    // same board type and medium.
+    $boys_key = $board_type . '|boys|' . $medium;
+    $girls_key = $board_type . '|girls|' . $medium;
+
+    if (in_array($boys_key, $education_levels_checked) || in_array($girls_key, $education_levels_checked)) {
+      return 'Conflict';
     }
   }
+  elseif (in_array($category, ['boys', 'girls'])) {
+    // Check if 'co-ed' already exists for the same board type and medium.
+    $coed_key = $board_type . '|co-ed|' . $medium;
+
+    if (in_array($coed_key, $education_levels_checked)) {
+      return 'Conflict';
+    }
+  }
+
+  // If no conflict is found, add the combination to the checked array.
+  $education_levels_checked[] = $key_for_conflict_check;
+  return 'Success';
```

Comment:
Instead of text, I would prefer numbers here, Like 0, 1, -1

Reply:
Ok.